File: modules/examples/src/main/java/edu/tum/cs/isabelle/examples/Hello_PIDE.java
package info.hupel.isabelle.examples.java;

import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Arrays;
import info.hupel.isabelle.api.*;
import info.hupel.isabelle.japi.*;
import info.hupel.isabelle.setup.*;

public class Hello_PIDE {

  public static void main(String args[]) {
    JResources res = JResources.dumpIsabelleResources();
    Configuration config = Configuration.simple("Protocol");
    Environment env = JSetup.makeEnvironment(JSetup.defaultSetup(new Version("2016")), res);
    JSystem sys = JSystem.create(env, config);
    String response = sys.invoke(Operations.HELLO, "world");
    System.out.println(response);
    sys.dispose();
  }

}

File: tools/hello_pide.py
from info.hupel.isabelle.api import Configuration, Version
from info.hupel.isabelle.japi import JResources, JSetup, JSystem, Operations

res = JResources.dumpIsabelleResources()
config = Configuration.simple("Protocol")
env = JSetup.makeEnvironment(JSetup.defaultSetup(Version.Stable("2017")), res)
sys = JSystem.create(env, config)
response = sys.invoke(Operations.HELLO, "world")
print response

sys.dispose()
